sdk-extensions/executor-service-processor/src/main/java/io/opentelemetry/sdk/extension/trace/export/ExecutorServiceSpanProcessor.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.trace.export;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import io.opentelemetry.sdk.trace.export.WorkerBase;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class ExecutorServiceSpanProcessor implements SpanProcessor {
+
+  private static final String SPAN_PROCESSOR_TYPE_LABEL = "spanProcessorType";
+  private static final String SPAN_PROCESSOR_TYPE_VALUE =
+      ExecutorServiceSpanProcessor.class.getSimpleName();
+
+  private final Worker worker;
+  private final AtomicBoolean isShutdown = new AtomicBoolean(false);
+  private final boolean ownsExecutorService;
+  private final ScheduledExecutorService executorService;
+  private final ScheduledFuture<?> future;
+
+  public static ExecutorServiceSpanProcessorBuilder builder(
+      SpanExporter spanExporter,
+      ScheduledExecutorService executorService,
+      boolean ownsExecutorService) {
+    return new ExecutorServiceSpanProcessorBuilder(
+        spanExporter, executorService, ownsExecutorService);
+  }
+
+  ExecutorServiceSpanProcessor(
+      SpanExporter spanExporter,
+      long scheduleDelayNanos,
+      int maxQueueSize,
+      int maxExportBatchSize,
+      long exporterTimeoutNanos,
+      ScheduledExecutorService executorService,
+      boolean ownsExecutorService,
+      long queuePeekMillis) {
+    this.worker =
+        new Worker(
+            spanExporter,
+            scheduleDelayNanos,
+            maxExportBatchSize,
+            exporterTimeoutNanos,
+            new ArrayBlockingQueue<>(maxQueueSize));
+    this.ownsExecutorService = ownsExecutorService;
+    this.executorService = executorService;
+    this.future =
+        executorService.scheduleAtFixedRate(
```

Comment:
I think it would be better to use scheduleWithFixedDelay to avoid getting in a busy loop if export takes too long. That may mean dropped spans, but this is IMHO usually preferable to taking too much CPU time away.